pkg/gardener/shoot/extender/dns.go

```diff
@@ -107,6 +108,16 @@ func NewDNSExtender(secretName, domainPrefix, dnsProviderType string) func(runti
 
 		shoot.Spec.Extensions = append(shoot.Spec.Extensions, dnsExtension)
 
+		secretReference := gardener.NamedResourceReference{
```

Comment:
I checked the validating webhook's [code](https://github.com/gardener/gardener-extension-shoot-dns-service/blob/master/pkg/admission/validator/shoot_validator.go#L57) once more. Now I'm sure we do the correct thing. Once the extension contains the secret reference the spec.resources field needs to be set accordingly.